XamlControlsGallery/ControlPages/NavigationViewPage.xaml

```diff
@@ -375,5 +379,56 @@
                 <local:ControlExampleSubstitution x:Name="navViewASB" Key="NavViewASB"/>
             </local:ControlExample.Substitutions>
         </local:ControlExample>
+
+        <local:ControlExample Grid.Row="10" 
+                              HeaderText="Hierarchical NavigationView"
+                              VerticalAlignment="Top" 
+                              HorizontalContentAlignment="Stretch" 
+                              XamlSource="NavigationView\NavigationViewSample8_xaml.txt"
+                              WebViewHeight="400">
+            <local:ControlExample.Options>
+                <StackPanel>
+                    <RadioButton x:Name="nvSample8Left" Content="Left mode" Checked="panePositionLeft_Checked" IsChecked="True"/>
+                    <RadioButton x:Name="nvSample8Top" Content="Top mode" Checked="panePositionTop_Checked"/>
+                    <RadioButton x:Name="nvSample8LeftCompact" Content="LeftCompact mode" Checked="panePositionLeftCompact_Checked"/>
+                </StackPanel>
+            </local:ControlExample.Options>
+            <Grid>
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="Auto"/>
+                    <RowDefinition Height="Auto"/>
+                </Grid.RowDefinitions>
+
+                <TextBlock TextWrapping="WrapWholeWords"
+                           Margin="0,0,0,12"
+                           Grid.Row="0"
+                           Text="NavigationView supports hierarchy in Left, LeftCompact, and Top display modes. Note that Left or Top mode works best if your parent items should navigate to specific pages, rather than just serve as headings. To expand the sub-menu without selecting the parent item, click the chevron. Choose between the three modes to see how each works." />
```

Comment:
Is there a mode wherein the parent page does NOT result in navigation? Should that be a separate sample?

Reply:
Yeah, it's possible that a top-level nav item doesn't actually go anywhere - just serves as a "heading" for its children sub-tree. I can change one of the top-level items in this sample so that it doesn't actually navigate anywhere - not sure if it warrants a whole other sample

Reply:
This added a little bit of confusion - why is one of the headers clickable and not the other? Maybe the text of the item could be updated to state which item has a page associated, and which is just a heading for its child sub tree? How is this documented in the guidance doc (could we use the same words here)?

Reply:
On that note: in LeftCompact mode, the user experience isn't optimal in the scenario where I'm already on Child A, and want to navigate to Child B. I need to navigate through Parent 1 even though I don't want to.

I guess this is more feedback about the feature itself. Do we have guidance docs for this scenario? CC: @Ranjeshj for FYI

Reply:
@stmoy do you mean arrow key navigation ? @ojhad fixed some keyboard navigation issues which should be in the latest pre-release related to arrow key navigation. Curious if you are trying with or without those changes.

Reply:
We have guidance docs currently in PR. But yeah, since left compact mode creates a flyout for the child menu, you have to navigate through the parent if you want to get to another child item.

I can update the headers to be more descriptive and try and show why one is just a header but the other one is clickable.

Reply:
From guidance doc:

When providing users with a hierarchical tree of navigation options, you may choose to make parent items non-selectable, for example when your app doesn't have a destination page associated with that parent item. To prevent an item from showing the selection indicator when invoked, set its SelectsOnInvoked property to False.

Reply:
After your comments and playing with the sample, I think I'm going to add this bit of guidance to the docs:

If your parent items are selectable, it's recommend you use the Left-Expanded or Top pane display modes. LeftCompact mode will cause the user to navigate to the parent item in order to open the child subtree every time its invoked.